src/Store/StoreThread.re

```diff
@@ -227,8 +242,15 @@ let start =
     // TODO: Wire this up properly
     let quickmenuSubs = quickmenuSubscriptionsUpdater(newState);
     QuickmenuSubscriptionRunner.run(~dispatch, quickmenuSubs);
-    let searchSubs = searchSubscriptionsUpdater(newState);
-    SearchSubscriptionRunner.run(~dispatch, searchSubs);
+
+    let searchDispatch = msg => dispatch(Model.Actions.Search(msg));
```

Comment:
With feature projects and a hierarchy we're going to be writing a lot of these wrapping function, so it's a good idea to try to make it as convenient as possible. There are a couple of ppxes that can compensate for the lack of first-class variant constructors to make this nicer:

```reason
// without ppxes and first-class vairant constructors
msg => dispatch(Msg.Search(msg))

// with ppx_pipebang (which inlines `|>`)
msg => msg |> Msg.Search |> dispatch

// with ppx_variants_conv and `>>` for function composition
dispatch >> Msg.search
```

Both of these are from Jane Street and require `base` during build, but fortunately not at runtime. Also, if we choose one, we might as well bring in the other one, and can have our pick at some other useful ones too.

Reply:
Cool! I've used `ppx_pipebang` before, it was pretty convenient, and I can confirm it works on Windows.